astropy/cosmology/flrw/tests/test_base.py

```diff
@@ -697,7 +697,7 @@ def test_Otot0(self, cosmo):
 
     @pytest.mark.skipif(not HAS_SCIPY, reason="scipy is not installed")
     @pytest.mark.parametrize("z, exc", invalid_zs)
-    @pytest.mark.parametrize("method", _FLRW_redshift_methods)
+    @pytest.mark.parametrize("method", sorted(_FLRW_redshift_methods))
```

Comment:
Why not just have `_FLRW_redshift_methods` be a list in the first place by applying the sort here?

https://github.com/astropy/astropy/blob/9ebf44a58bf36ec69989dbc790e8bea613cd59d7/astropy/cosmology/tests/helper.py#L78

Reply:
I initially tried that, but the generated set is modified in a number of tests where redshift methods are removed.  See the rest of the diff.  Turning the resulting set into a sorted list was the simplest solution, and it is done elsewhere in the code base.

https://github.com/astropy/astropy/blob/a3cd9a9983badbd600987bcabc31fd67a3f21e6d/astropy/timeseries/periodograms/lombscargle/tests/test_statistics.py#L119

Reply:
I agree that maybe these tests are a little over-engineered but that is neither here nor there. Let's get this in as-is then. Really appreciate your help. Thanks!